examples/apps/generic_client/proxy_mode_generic_client.py

```diff
@@ -0,0 +1,377 @@
+            "-w", "--worker_id",
+            help="worker id in plain text to use to submit a work order",
+            type=str)
+        mutually_excl_group_worker.add_argument(
+            "-wx", "--worker_id_hex",
+            help="worker id as hex string to use to submit a work order",
+            type=str)
+        parser.add_argument(
+            "-l", "--workload_id",
+            help='workload id for a given worker',
+            type=str)
+        parser.add_argument(
+            "-i", "--in_data",
+            help='Input data',
+            nargs="+",
+            type=str)
+        parser.add_argument(
+            "-o", "--decrypted_output",
+            help="If present, display decrypted output as JSON",
+            action='store_true')
+        parser.add_argument(
+            "-rs", "--requester_signature",
+            help="Enable requester signature for work order requests",
+            action="store_true")
+        options = parser.parse_args(args)
+
+        return options
+
+    def _parse_config_file(self, config_file):
+        # Parse config file and return a config dictionary.
+        conf_files = [config_file]
+        confpaths = ["."]
+        try:
+            config = pconfig.parse_configuration_files(conf_files, confpaths)
+            json.dumps(config)
+        except pconfig.ConfigurationException as e:
+            logger.error(str(e))
+            config = None
+
+        return config
+
+    def is_requester_signature_enabled(self):
+        return self._requester_signature
+
+    def is_decrypt_response_enabled(self):
+        return self._show_decrypted_output
+
+    def get_first_active_worker(self):
+        if self._worker_id:
+            return self._worker_id
+        # Get first worker id from worker registry
+        worker_id = None
+        w_count, _, worker_ids = self._worker_instance.worker_lookup(
+            WorkerType.TEE_SGX,
+            self._config["WorkerConfig"]["OrganizationId"],
+            self._config["WorkerConfig"]["ApplicationTypeId"]
+        )
+        logger.info("\n Worker lookup response: count {} "
+                    "worker ids {}\n".format(
+                        w_count, worker_ids))
+
+        if w_count != 0:
+            # Iterate through all workers and find get
+            # first active worker.
+            for w_id in worker_ids:
+                wo_ret_result = self._worker_instance.worker_retrieve(w_id)
+                # First argument indicates status of worker.
+                if wo_ret_result[0] == WorkerStatus.ACTIVE.value:
+                    worker_id = w_id
+                    break
+            return worker_id
+        else:
+            logger.error("ERROR: Failed to lookup worker")
+            sys.exit(-1)
+
+    def create_work_order_params(self, worker_id, worker_encrypt_key,
+                                 session_key, session_iv,
+                                 verification_key):
```

Comment:
Can we keep common functions used by proxy and direct model clients in a separate file ? That way both direct/proxy model clients can reuse the functions.

Reply:
Don't compare it with direct model right now. I have one more PR to raise for that.. I will create 2 classes for direct and proxy model and common code also.